### PR TITLE
Implement writeable context for observations

### LINKED ISSUE #103: make image, weight etc. refs of observations read only by default
add a context manger that switches on writeable refs and runs update_pixels on exit

this brings up again the issue of jacobians, which are always copies in the current code

### PR DESCRIPTION
**Pull request by esheldon**
closes #103 

Update the metacal addnoise code to use the context

### CODE REVIEW

**Comment by esheldon** [comment]
I don't think we require it for the other images, they are not stored in the pixels array

**Comment by beckermr** [comment]
Sure but the api is pretty fractured then.

**Comment by esheldon** [comment]
after more thought I agree.

But while we are at it we should fix the inconsistency of the jacobian, which returns a copy.  This will definitely break some code

**Comment by esheldon** [comment]
ok to merge?

File: ngmix/bootstrap.py [hunk 1]
```diff
@@ -279,93 +274,6 @@ def _get_round_pars(self, pars_in):
 
         return pars, pars_lin
 
-
-    def _find_cen(self, ntry=10, Tguess=4.0):
```
Comment (beckermr) [inline ngmix/bootstrap.py]: What happened to this method? Do we not need it anymore?

Comment (esheldon) [reply ngmix/bootstrap.py]: no, it was never used in production

**Pull request merged**